libgadget/run.c

```diff
@@ -249,6 +253,8 @@ void run(void)
         /* Update velocity to the new step, with the newly computed step size */
         apply_half_kick();
 
+        free_activelist();
```

Comment:
Why is activelist no longer needed after this?

Perhaps shall pass-in ActiveParticles as an argument to these functions above to signify if they need it.